server/utils/firebase_algolia_server_code/functions/test/index.test.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Unit tests for index.js
+ * Uses dependency injection to mock algolia save/delete functions
+ * Uses sinon/chai to spy on calls and validate behavior
+ * compile with `npm run test`
+ * Having problems running this on your local? 
+ * - Download a firebase service key.
+ * - *DON'T* push this key to github please.  
+ */
+
+// Initialize the firebase test mocking environment.
+const test = require('firebase-functions-test')({
+  databaseURL: 'https://step9-2020-capstone.firebaseio.com',
+  storageBucket: 'step9-2020-capstone.appspot.com',
+  projectId: 'step9-2020-capstone',
+}, 'step9-2020-capstone.json');
+
+// Initialize chai, sinon, sinon-chai. 
+var assert = require('chai').assert;
+var chai = require("chai");
+var sinon = require("sinon");
+var sinonChai = require("sinon-chai");
+chai.should();
+chai.use(sinonChai);
+
+// Import the generic cloud functions.
+const {createRecordFromEntity, updateRecordInIndex, deleteEntityFromIndex} = 
+    require('../index.js');
+
+// Import mock saveObject, mock deleteObject, and helpers.
+const {saveObject, deleteObject} = require('./algoliaMocks');
+const {getFormattedSnap, getFormattedChange} = require('./exampleDataHelpers');
+
+describe("test_createRecordFromEntity", () => {
+  it('createRecordFromEntity returns exactly the input data', () => {
+    const wrappedAdd = test.wrap(createRecordFromEntity(saveObject));
+
+    const snap = test.firestore.exampleDocumentSnapshot();
+    const addedSnap = wrappedAdd(snap);
+
+    assert.deepEqual(addedSnap, getFormattedSnap(snap));
+  });
+
+  it('createRecordFromEntity calls saveObject exactly once', () => {
+    var saveSpy = sinon.spy(saveObject);
```

Comment:
Conventionally I think `spy` is good enough for a name, and also change `var` to `const` please